### PR TITLE
Rumpkernel

### PR DESCRIPTION
**Pull request by RobertGiff**
### Summary of this Pull Request (PR)

Update Composite installation guide with necessary Linux packages.

### Intent for your PR

Choose one (Mandatory):

- [x] This PR is for a code-review and is intended to get feedback, but not to be pulled yet.
- [ ] This PR is mature, and ready to be integrated into the repo.

### Reviewers (Mandatory):
@gparmer @maloneya @Others 

### Code Quality

As part of this pull request, I've considered the following:

[Style](https://github.com/gparmer/composite/raw/ppos/doc/style_guide/composite_coding_style.pdf):

- [x] Comments adhere to the Style Guide (SG)
- [x] Spacing adhere's to the SG
- [x] Naming adhere's to the SG
- [x] All other aspects of the SG are adhered to, or exceptions are justified in this pull request
- [x] I have run the auto formatter on my code before submitting this PR (see doc/auto_formatter.md for instructions)

[Code Craftsmanship](http://www2.seas.gwu.edu/~gparmer/posts/2016-03-07-code-craftsmanship.html):

- [x] I've made an attempt to remove all redundant code
- [x] I've considered ways in which my changes might impact existing code, and cleaned it up
- [x] I've formatted the code in an effort to make it easier to read (proper error handling, function use, etc...)
- [x] I've commented appropriately where code is tricky
- [x] I agree that there is no "throw-away" code, and that code in this PR is of high quality

### Testing
N/A

### CODE REVIEW

**Comment by Others** [comment]
@RobertGiff for the Rust instructions, you can remove the temporary file and just pipe it to sh (you only need the temporary script if you aren't willing to answer shell prompts)

File: doc/README.md [hunk 1]
```diff
@@ -1,5 +1,24 @@
 # Composite Setup Guide
 
+## Tools nessesary for Composite
```
Comment (phanikishoreg) [inline doc/README.md]: "necessary"

File: doc/README.md [hunk 2]
```diff
@@ -1,5 +1,24 @@
...
+```
+sudo apt-get -y install bc  
+sudo apt-get -y install gcc-multilib
```
Comment (phanikishoreg) [inline doc/README.md]: We don't beed `multilib` for 32bit. should we note that?

File: doc/README.md [hunk 3]
```diff
@@ -1,5 +1,24 @@
 # Composite Setup Guide
 
+## Tools nessesary for Composite
+```
+sudo apt-get -y install bc  
+sudo apt-get -y install gcc-multilib  
+sudo apt-get -y install binutils-dev  
+sudo apt-get -y install qemu-kvm
```
Comment (phanikishoreg) [inline doc/README.md]: I always install `qemu` vs `qemu-kvm` here. Does it matter?

File: doc/README.md [hunk 4]
```diff
@@ -1,5 +1,24 @@
...
 
+## Tools nessesary for Composite
+```
+sudo apt-get -y install bc  
+sudo apt-get -y install gcc-multilib  
+sudo apt-get -y install binutils-dev  
+sudo apt-get -y install qemu-kvm  
+sudo apt-get -y install g++  
+sudo apt-get -y install make  
+sudo apt-get -y install cmake
```
Comment (phanikishoreg) [inline doc/README.md]: Also, `sudo apt-get install build-essential` should take care of `gcc`, `g++`, `make`, which is what I usually install. Package info: https://packages.ubuntu.com/trusty/devel/build-essential.

File: doc/README.md (hunk 2 above)
Comment (gparmer) [reply doc/README.md]: Doesn't really hurt to include it ;-)

**Comment by gparmer** [comment]
FYI, some information about the PR template:

The "- [ x]" should be "- [x]"  https://github.com/blog/1825-task-lists-in-all-markdown-documents

Under testing, you normally include which of the tests (e.g. `micro_booter`) you ran.  That doesn't apply here as you're adding documentation, so "N/A" is fine.

**Comment by phanikishoreg** [comment]
@robertgiff we also need grub and iso utilities to make a composite bootable iso image, right?

File: doc/README.md (hunk 3 above)
Comment (RobertGiff) [reply doc/README.md]: https://www.quora.com/Virtualization-What-is-the-difference-between-KVM-and-QEMU

**Comment by RobertGiff** [comment]
@phanikishoreg making a composite bootable iso I believe just requires package grub-common to run command `grub-mkrescue`. I'll post more if I find more to be required.

**Comment by phanikishoreg** [comment]
`grub-mkrescue` depends on `xoriso` and that is not part of `grub-common`.

**Comment by RobertGiff** [comment]
Interesting, when I run `dpkg -S grub-mkrescue` to get the command's package I get grub-common. I assume xorriso also installs it?

**Comment by phanikishoreg** [comment]
Perhaps `xoriso` maintainers are different, could be why `grub-common` doesn't include it. 
All I know is `grub-mkrescue` requires `xorriso` and we need to install them separately. ;-)

File: doc/README.md [hunk 5]
```diff
@@ -1,5 +1,21 @@
...
+sudo apt-get -y install binutils-dev  
+sudo apt-get -y install qemu-kvm  
+sudo apt-get -y install cmake  
+sudo apt-get -y install build-essential  
+sudo apt-get -y install grub-common  
+  
+curl https://sh.rustup.rs -sSf | sh 
+  
+. ~/.cargo/env
+rustup component add rust-src
```
Comment (Others) [inline doc/README.md]: Need a line before this: `rustup default nightly`

File: doc/README.md [hunk 6]
```diff
@@ -1,5 +1,21 @@
...
+sudo apt-get -y install gcc-multilib  
+sudo apt-get -y install binutils-dev  
+sudo apt-get -y install qemu-kvm  
+sudo apt-get -y install cmake  
+sudo apt-get -y install build-essential  
+sudo apt-get -y install grub-common  
+  
+curl https://sh.rustup.rs -sSf | sh 
+  
+. ~/.cargo/env
```
Comment (Others) [inline doc/README.md]: Personally I think `source ~/.cargo/env` is cleaner

**Comment by gparmer** [comment]
Thanks @RobertGiff !

**Pull request merged**